sources/PTYSession.m

```diff
@@ -2635,14 +2635,19 @@ - (void)setPreferencesFromAddressBookEntry:(NSDictionary *)aePrefs
         _tmuxTitleOutOfSync = YES;
     }
 
+    BOOL use_underline = [iTermProfilePreferences boolForKey:KEY_USE_UNDERLINE_COLOR inProfile:aDict];
+
     NSDictionary *keyMap = @{ @(kColorMapForeground): KEY_FOREGROUND_COLOR,
                               @(kColorMapBackground): KEY_BACKGROUND_COLOR,
                               @(kColorMapSelection): KEY_SELECTION_COLOR,
                               @(kColorMapSelectedText): KEY_SELECTED_TEXT_COLOR,
                               @(kColorMapBold): KEY_BOLD_COLOR,
                               @(kColorMapLink): KEY_LINK_COLOR,
                               @(kColorMapCursor): KEY_CURSOR_COLOR,
-                              @(kColorMapCursorText): KEY_CURSOR_TEXT_COLOR };
+                              @(kColorMapCursorText): KEY_CURSOR_TEXT_COLOR,
+                              @(kColorMapUnderline): (use_underline ? KEY_UNDERLINE_COLOR : [NSNull null])
+                              };
+
     for (NSNumber *colorKey in keyMap) {
         NSString *profileKey = keyMap[colorKey];
         NSColor *theColor = [[iTermProfilePreferences objectForKey:profileKey
```

Comment:
`objectForKey:inProfile:` expects an NSString* for the first argument, and passing an `NSNull` works only by accident. Better to say

````
NSColor *theColor = nil;
if ([profileKey isKindOfClass:[NSString class]]) {
  theColor = ...
}
````